nixos/modules/virtualisation/amazon-init.nix

```
@@ -12,11 +12,26 @@ let
 
     userData=/etc/ec2-metadata/user-data
 
+    isScript() {
+        local fn="$1"
+        local fd
+        local magic
+        exec {fd}< "$fn"
+        read -r -n 2 -u "$fd" magic
+        exec {fd}<&-
+        if [[ "$magic" =~ \#! ]]; then return 0; else return 1; fi
```

Doing some iterating/golfing:

Test already returns 0/1.
```
        return [[ "$magic" =~ \#! ]]
```

Function returns result of last command.
```
        [[ "$magic" =~ \#! ]]
```

We're only testing for exact match, no need for regex. We're only reading 2 chars into the var.
```
        [[ "$magic" == \#! ]]
```

POSIX
```suggestion
        [ "$magic" = \#! ]
```